GradientLoadingBar/Classes/GradientLoadingBarController.swift

```diff
@@ -116,6 +116,20 @@ open class GradientLoadingBarController {
                                               completion: completion)
     }
 
+    /// Progress display through rate ---- Rate range : 0 ~ 1
+    public func setProgress(_ rate: CGFloat) {
+        guard rate >= 0, rate <= 1 else { return }
+
+        guard let window = viewModel.superview.value, let width = window?.frame.size.width else { return }
+
+        let progress = width * rate
+        let widthConstraints = gradientActivityIndicatorView.constraints.filter { $0.firstAttribute == .width }
+        guard !widthConstraints.isEmpty else { return }
+
+        let widthConstraint = widthConstraints[0]
+        widthConstraint.constant = progress
+    }
```

Comment:
I would prefer having this logic on a subclass, as there are cases where the width doesn't reflect the progress, e.g. here https://github.com/fxm90/GradientLoadingBar/pull/19